src/gui/sdlmain.cpp

```diff
@@ -2229,6 +2241,71 @@ static SDL_Point window_bounds_from_label(const std::string &pref,
 	return {w, h};
 }
 
+static SDL_Point clamp_to_minimum_window_dimensions(SDL_Point size)
+{
+	const auto w = std::max(size.x, FALLBACK_WINDOW_DIMENSIONS.x);
+	const auto h = std::max(size.y, FALLBACK_WINDOW_DIMENSIONS.y);
+	return {w, h};
+}
+
+static SDL_Rect get_desktop_resolution()
+{
+	SDL_Rect desktop;
+	assert(sdl.display_number >= 0);
+	SDL_GetDisplayBounds(sdl.display_number, &desktop);
+	assert(desktop.w >= FALLBACK_WINDOW_DIMENSIONS.x);
+	assert(desktop.h >= FALLBACK_WINDOW_DIMENSIONS.y);
+	return desktop;
+}
+
+// Takes in:
+//  - The 'maxresolution' config value: 'default', 'W,H', or an invalid setting.
+//
+// Except for SURFACE and TEXTURE rendering, the function populates the following struct members:
+//  - 'sdl.desktop.restrict_resolution', true if resolution restriction is enabled.
+//  - 'sdl.desktop.max_resolution', with the refined size.
+
+static void setup_resolution_restriction_from_conf(const char *max_resolution_val)
+{
+	// TODO: Deprecate SURFACE output and remove this.
+	if (sdl.desktop.want_type == SCREEN_SURFACE)
+		return;
+
+	// TODO: Implement support for TEXTURE output.
+	if (sdl.desktop.want_type == SCREEN_TEXTURE)
+		return;
```

Comment:
From a users perspective, the `[sdl]` resolution settings are meant to be general - regardless of what output method they're using.

I know we're in full agreement we want TEXTURE supported here too - but in terms of merging this, I am extremely reluctant to accept it without TEXTURE support (as many users exclusively run in this mode.. such as on ARM SBCs like the Pi).

Reply:
Ideally this function would be agnostic of underlying output technology.

It's description implies it takes in the user's setting and returns (or sets) the general max-resolution numbers, after which the output mechanisms can do the right thing with those numbers.

If you look at the `setup_window_sizes_from_conf()` function - what it does and where it's called: ideally we could do the same here, in an output-agnostic way?

Reply:
Okay, I'll add texture support, I shouldn't be that difficult. I get your point that it should be output-agnostic, I'll just set the defaults that effectively turn it off for SURFACE instead.